workspace_host/interface.js

```diff
@@ -1397,19 +1398,31 @@ async function sendGradedFilesArchive(workspace_id, res) {
     workspaceDir = path.join(config.workspaceHostHomeDirRoot, workspace.remote_name, 'current');
   }
 
+  const gradedFiles = await fg(workspaceSettings.workspace_graded_files, {
+    cwd: workspaceDir,
+    stats: true,
+  });
+
+  if (gradedFiles.length > config.workspaceMaxGradedFilesCount) {
+    return res.status(500).send({ message: 'Too many files submitted in workspace.' });
+  }
+  if (_.sumBy(gradedFiles, (file) => file.stats.size) > config.workspaceMaxGradedFilesSize) {
+    return res.status(500).send({ message: 'Workspace files are larger than allowed.' });
+  }
```

Comment:
I'd consider including the actual limits here, e.g. "Cannot submit more than ### files from workspace" and "Workspace files total size exceeded ###". We've used https://www.npmjs.com/package/filesize elsewhere to get a human-readable string for a number of bytes.

Reply:
Fixed in 8425f6166.